sal/class.json-api-site-jetpack.php

```diff
@@ -139,7 +139,25 @@ function allowed_file_types() {
 	}
 
 	function is_private() {
-		return false;
+		return $this->get_cloud_site_option( 'blog_public' ) == -1;
+	}
+
+	function is_coming_soon() {
+		return $this->is_private() && $this->get_cloud_site_option( 'wpcom_coming_soon' ) == 1;
+	}
+
+	function get_cloud_site_option( $option ) {
+		$jetpack = Jetpack::init();
+		if ( ! method_exists( $jetpack, 'get_cloud_site_options' ) ) {
+			return false;
+		}
+
+		$result = $jetpack->get_cloud_site_options( [ $option ] );
```

Comment:
It requires one update on API side. Also, I'd like to only make this call on atomic sites and not on regular Jetpack sites, so this PR needs a bit more attention.